### PR TITLE
Remove legacy ngraph builders

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Remove legacy ngraph builders
 - Remove ngraph test utils

### Tickets:
 - [CVS-131962](https://jira.devtools.intel.com/browse/CVS-131962)

### CODE REVIEW

File: src/tests/ov_helpers/ov_models/include/ov_models/utils/ov_helpers.hpp [hunk 1]
```diff
@@ -18,91 +18,6 @@
...
-template <class opType>
-inline ov::NodeVector castOps2Nodes(const std::vector<std::shared_ptr<opType>>& ops) {
-    ov::NodeVector nodes;
-    for (const auto& op : ops) {
-        nodes.push_back(std::dynamic_pointer_cast<ov::Node>(op));
-    }
-    return nodes;
-}
-
 std::vector<std::pair<ov::element::Type, std::vector<std::uint8_t>>> interpreterFunction(
```
Comment (olpipi) [inline src/tests/ov_helpers/ov_models/include/ov_models/utils/ov_helpers.hpp:21]: To be removed in the next PR

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/range.cpp [hunk 2]
```diff
@@ -4,8 +4,7 @@
 //
 //#include "utils/cpu_test_utils.hpp"
 //
-////#include "ov_models/utils/ov_helpers.hpp"
-//
+//////
```
Comment (praasz) [inline src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/range.cpp:7]: Can be removed?

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache.cpp [hunk 3]
```diff
@@ -99,6 +99,19 @@ const std::vector<std::vector<InputShape>> input_shapes_basic = {
     },
 };
 
+void resize_function(std::shared_ptr<ov::Model> function, const std::vector<ov::Shape>& targetInputStaticShapes) {
+    auto inputs = function->inputs();
+    std::map<ov::Output<ov::Node>, ov::PartialShape> shapes;
+    if (inputs.size() > targetInputStaticShapes.size()) {
```
Comment (praasz) [inline src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache.cpp:105]: As this is test code use gtest assertion ASSERT_GT() << "error message"; ?

File: src/tests/test_utils/common_test_utils/src/ov_test_utils.cpp [hunk 4]
```diff
@@ -108,3 +110,22 @@ void check_unique_names(const std::shared_ptr<ov::Model>& f, const std::shared_p
 std::shared_ptr<ov::op::v0::Constant> create_zero_constant(const ov::element::Type_t& et, const ov::Shape& shape) {
     return ov::op::v0::Constant::create(et, shape, {0});
 }
+
+namespace ov {
+namespace test {
+namespace utils {
+
+bool is_tensor_iterator_exist(const std::shared_ptr<ov::Model>& func) {
```
Comment (praasz) [inline src/tests/test_utils/common_test_utils/src/ov_test_utils.cpp:118]:
```suggestion
bool is_tensor_iterator_exist(const std::shared_ptr<ov::Model>& model) {
```

File: src/tests/test_utils/common_test_utils/include/common_test_utils/ov_test_utils.hpp [hunk 5]
```diff
@@ -71,3 +71,13 @@ std::shared_ptr<ov::op::v0::Constant> create_constant(const std::vector<T>& data
 }
 
 std::shared_ptr<ov::op::v0::Constant> create_zero_constant(const ov::element::Type_t& et, const ov::Shape& shape);
+
+namespace ov {
+namespace test {
+namespace utils {
+
+bool is_tensor_iterator_exist(const std::shared_ptr<ov::Model>&);
```
Comment (praasz) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/ov_test_utils.hpp:79]:
```suggestion
bool is_tensor_iterator_exist(const std::shared_ptr<ov::Model>& model);
```
Is required to pass reference to shared pointer maybe use just ov::Model

```suggestion
bool is_tensor_iterator_exist(const ov::Model& model);
```

**Review by vurusovs — APPROVED** [review]

**Comment by olpipi** [comment]
is blocked by
https://github.com/openvinotoolkit/openvino_contrib/pull/871
https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8901

File: src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/range.cpp (hunk 2 above)
Comment (olpipi) [reply src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/range.cpp]: done

File: src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache.cpp (hunk 3 above)
Comment (olpipi) [reply src/plugins/intel_gpu/tests/functional/subgraph_tests/dynamic/kv_cache.cpp]: This is a free function not related to direct code execution. So I would not use gtest macroces.
But there is no reason to use std::runtime_error. So I changed it to OPENVINO_ASSERT

File: src/tests/test_utils/common_test_utils/src/ov_test_utils.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/src/ov_test_utils.cpp]: done

File: src/tests/test_utils/common_test_utils/include/common_test_utils/ov_test_utils.hpp (hunk 5 above)
Comment (olpipi) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/ov_test_utils.hpp:79]: Usually we operate with std::shared_ptr<ov::Model> not with ov::Model. So it is just more convenient to use this function with shared_ptr

**Review by praasz — APPROVED** [review]

**Pull request merged**